policy/nomad/source.go

```diff
@@ -102,41 +104,26 @@ func (s *Source) MonitorIDs(ctx context.Context, req policy.MonitorIDsReq) {
 		// Perform a blocking query on the Nomad API that returns a stub list
 		// of scaling policies. The call is done in a goroutine so we can
 		// still listen for the context closing or a reload request.
-		blockingQueryCompleteCh := make(chan struct{})
-		go func() {
-			// Obtain a handler now so we can release the lock before starting
-			// the blocking query.
-			s.nomadLock.RLock()
-			scaling := s.nomad.Scaling()
-			s.nomadLock.RUnlock()
+		//blockingQueryCompleteCh := make(chan struct{})
 
-			policies, meta, err = scaling.ListPolicies(q)
-			close(blockingQueryCompleteCh)
-		}()
+		// Obtain a handler now so we can release the lock before starting
+		// the blocking query.
+		s.nomadLock.RLock()
+		scaling := s.nomad.Scaling()
+		s.nomadLock.RUnlock()
 
-		select {
-		case <-ctx.Done():
-			s.log.Trace("stopping ID subscription")
-			return
-		case <-s.reloadCh:
-			s.log.Trace("reloading policies")
-			continue
-		case <-blockingQueryCompleteCh:
-		}
```

Comment:
if we want to clean this up just a little bit, this channel could instead be a `make(chan error)`, and send the `ListPolicies` `err` down it. that way the next `err != nil` check wouldn't be referring to something set directly in the goroutine.  something like

```go
case err = <-listPoliciesError:
```

Reply:
This part of the code will have a lot of changes, so I didn't want to do much refactoring just yet.